### ISSUE TITLE
Cargo needs are not always correct

### ISSUE THREAD

**Issue by Tkael** [EDCD/EDDI#2645]
## What's Wrong (please be as specific as possible)
Cargo needs are not always correct and in some cases are zero when they ought not be.

### Expected
Cargo needs are always correct when requested

### Observed
Cargo needs are not always correct and in some cases are zero when they ought not be.

### Steps to reproduce
1. [Give the fullest and most reproducible steps you can]
2. [The more reproducible, the better our chances of fixing it]

### Configuration
* Version
  * EDDI Version: 4.1.0-b3
  * VoiceAttack Version: N/A
* Localization
  * EDDI's language: N/A
  * Voice: N/A

### Investigation Notes
CalculateCargoNeeds() does not run or update after collecting, refining, or otherwise acquiring cargo.

### EDDI Logs
[Any log files providing additional details about what happened. These are located at %APPDATA%/EDDI and have the extension ".log"]

### Player journals
[These are not required if EDDI Logs are verbose but may be helpful otherwise. They are located at [your Saved Games folder]/Frontier Developments/Elite Dangerous/. Please only provide excerpts or attach files from the play session where the issue occurred.]

**Comment by Tkael** [comment]
Per Darkcyde:
Just running another Rescue (surface salvage) mission.  When I picked it up at the station, I check the `.need` and it correctly stated 4.  I've now picked up a couple at the mission location, and checked again, but now it's saying zero.  So it's been wiped somewhere in between taking it and being at the location.  I'll see if I can do some more and check at various stages to see if i can work out where it's being wiped.

**Comment by Darkcyde13** [comment]
Just thought I'd mention this.  While running tests for the Surface Signals/Locations execution order on game start, I noticed that after a restart of EDDI, that the cargo.need is now zero.  I guess it needs to be set/updated when you load into a game too.

EDIT: Sorry, Ignore that.  I forgot to test for `.need` after I collected the commodity and before the SS/L testing, so it could have been wiped before the testing.  I'll try to run some more to clarify.
